### PR TITLE
update install script to include setting ACL grants for typical linux log directories

### PR DESCRIPTION
**Pull request by jspaleta**
This is addresses the need to provide the sumo-otelcol permissions to read log files without having to manipulate multiple user or group definitions.

_most_ linux distributions _should_ have acl support enabled by default on their mounted filesystems...but its not universal.  This needs to be tested across officially supported linux distribution target configurations.

This PR assumes includes code to test for availability of the setfacl command, and if not found will skip the ACL manipulation.

### CODE REVIEW

File: scripts/install.sh [hunk 1]
```diff
@@ -855,6 +856,14 @@ function get_binary_from_url() {
     fi
 }
 
+function set_acl_on_log_paths() {
+for path in ${LOG_FILE_PATHS}; do
+    echo -e "Running: setfacl -R -m d:u:${SYSTEM_USER}:r-x,u:${SYSTEM_USER}:r-x,g:${SYSTEM_USER}:r-x ${path}"
+    setfacl -R -m d:u:${SYSTEM_USER}:r-x,u:${SYSTEM_USER}:r-x,g:${SYSTEM_USER}:r-x ${path}
```
Comment (sumo-drosiek) [inline scripts/install.sh:862]: is `setfacl` command always installed? If not please handle scenario when its missing

Comment (jspaleta) [reply scripts/install.sh]: setfacl command conditionally tested for in latest commit

**Comment by sumo-drosiek** [comment]
Sorry for additional asks. Could init variable on the [top of the script](https://github.com/SumoLogic/sumologic-otel-collector/blob/cde83fff401ae898c7528805fc69a2e160b8d5a3/scripts/install.sh#L58-L77) and make it readonly after [setting defaults](https://github.com/SumoLogic/sumologic-otel-collector/blob/cde83fff401ae898c7528805fc69a2e160b8d5a3/scripts/install.sh#L926-L927)?

**Review by sumo-drosiek — APPROVED** [review]

**Comment by jspaleta** [comment]
@sumo-drosiek 
Do the install related tests need to be updated to make sure the setfacl command is installed so this new codepath gets tested?

**Comment by sumo-drosiek** [comment]
I believe so. Do not require it in this PR, but if change is not urgent tests would be very helpful

**Comment by swiatekm-sumo** [comment]
@jspaleta can you resolve conflicts on this? Looks like we're good to go otherwise.

**Review by sumo-drosiek — COMMENTED** [review]
Could we add tests for it?

**Comment by portertech** [comment]
We could add tests for it, but it would likely require `sudo` or `su` and a consistent existing log file to test the users access. Alternatively, we could run the ACL tool to inspect the ACL on `/var/log`.

**Comment by portertech** [comment]
I don't believe ACL info will be represented in https://pkg.go.dev/io/fs#FileMode, I'll test and tinker.

**Pull request merged**